### ISSUE TITLE
CrateDB 5.8.4 no longer accepts `ARRAY`s with varying inner types in `OBJECT(IGNORED)` columns

### ISSUE THREAD

**Issue by amotl** [crate/crate#16734]
### CrateDB version

5.8.4

### CrateDB setup information

Standard Docker.
```
docker run -it --rm --publish 4200:4200 --env CRATE_HEAP_SIZE=2g crate/crate:5.8.4 crate -Cdiscovery.type=single-node
```

### Problem description

While it still works in 5.8.3, when using CrateDB 5.8.4, it no longer accepts storing `ARRAY`s like `[42.42, "foo"]`. We are using that capability of CrateDB when [importing data from DynamoDB](https://cratedb-toolkit.readthedocs.io/io/dynamodb/), where we are deferring data into a `OBJECT(IGNORED)` column, that can not be stored into a `OBJECT(DYNAMIC)` column.

### Steps to Reproduce

```sql
DROP TABLE IF EXISTS t01;
CREATE TABLE IF NOT EXISTS t01 (data OBJECT(IGNORED));
INSERT INTO t01 (data) VALUES ('{"items": [42.42, "foo"]}'::json);
REFRESH TABLE t01;
SELECT * FROM t01;
```

### Actual Result

```python
SQLParseException[Mixed dataTypes inside a list are not supported. Found double precision and text]
```

### Expected Result

```sql
INSERT OK, 1 row affected (0.008 sec)
```
```
# SELECT * FROM t01;

+---------------------------+
| data                      |
+---------------------------+
| {"items": [42.42, "foo"]} |
+---------------------------+
```

**Comment by BaurzhanSakhariev** [comment]
Hi @amotl, thanks for catching! It was a regression introduced in 5.8.4 fix will be available in the next 5.8.5 hotfix release.

**Comment by amotl** [comment]
Thank you for providing a quick fix!
